Sources/ArcGISToolkit/FloorFilter/SiteFacilityPromptViewController.swift

```diff
@@ -0,0 +1,272 @@
+import ArcGIS
+import Foundation
+
+/// ViewController for the site and facility prompt
+internal class SiteFacilityPromptViewController: UIViewController {
+    var delegate: FloorFilterViewControllerDelegate?
+    var viewModel = FloorFilterViewModel()
+    
+    /// UI Elements and constraints
+    @IBOutlet weak var siteFacilitySearchBar: UISearchBar!
+    @IBOutlet weak var backBtn: UIImageView!
+    @IBOutlet weak var closeBtn: UIImageView!
+    @IBOutlet weak var promptTitle: UILabel!
+    @IBOutlet weak var promptSubtitle: UILabel!
+    @IBOutlet weak var promptTitleSubtitleStackView: UIStackView!
+    @IBOutlet weak var siteFacilityTableView: UITableView!
+    @IBOutlet weak var bottomConstraint: NSLayoutConstraint!
+    @IBOutlet weak var designableViewHeight: NSLayoutConstraint!
+    
+    private var originalYPositionForPanel: CGFloat = 0.0
+    
+    /// Show the facilities list directly if the map has no sites configured or if there is a previously selected facility
+    private var isShowingFacilities = false
+    private var isSearchActive = false
+    
+    /// Filtered facilities and sites list based on search query
+    private var filteredSearchFacilities: [AGSFloorFacility] = []
+    private var filteredSearchSites: [AGSFloorSite] = []
+    
+    private func getFilteredFacilities() -> [AGSFloorFacility] {
+        return isSearchActive ? filteredSearchFacilities : viewModel.facilities
+    }
+    
+    private func getFilteredSites() -> [AGSFloorSite] {
+        return isSearchActive ? filteredSearchSites : viewModel.sites
+    }
+    
+    override public func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+        isShowingFacilities = (viewModel.sites.isEmpty || viewModel.selectedFacility != nil) ? true : false
+        initializeButtonsClickListeners()
+        updatePromptViewHeight()
+        updatePromptTitle()
+    }
+    
+    override public func viewDidLoad() {
+        super.viewDidLoad()
+        initializeSiteFacilityTableView()
+        initializeSiteFacilitySearchBar()
+        initializeButtonsClickListeners()
+       
+        updatePromptViewHeight()
+        updatePromptTitle()
+    }
+    
+    private func initializeButtonsClickListeners() {
+        closeBtn.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.closeSiteFacilityPrompt)))
+        closeBtn.isUserInteractionEnabled = true
+        backBtn.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(self.backButtonPressed)))
+        backBtn.isUserInteractionEnabled = true
+    }
+    
+    private func updatePromptViewHeight() {
+        self.designableViewHeight.constant = 380
+        bottomConstraint.constant = ((UIScreen.main.bounds.height / 2) - (self.designableViewHeight.constant / 2))
+        originalYPositionForPanel = bottomConstraint.constant
+    }
+    
+    @objc func closeSiteFacilityPrompt() {
+        self.dismiss(animated: true, completion: nil)
```

Comment:
`completion` has a default value of `nil`.
```suggestion
        dismiss(animated: true)
```